specification.md

```diff
@@ -142,22 +142,232 @@ An ingredient group is a group of related ingredients, e.g. the
 ingredients making up one component of a dish. It consists of:
 
 - *optional* a title
-- 1..n ingredients
+- 0..n ingredients
+- 0..n ingredients groups
 
 An ingredient group us represented as follows:
 
 1. A [heading], whose contents are the group's title
-2. A list of ingredients
-
-[heading]: https://spec.commonmark.org/0.28/#atx-headings
-
-
+2. A list of ingredients (can be skipped)
+3. A group may have child groups: Any groups directly following the
+   group that have a heading level greater than the group's heading
+   are parsed as children.
```

Comment:
I think the wording “heading level greater than” can be misleading in this case. I couldn't find something about this in the commonmark spec, but e.g. [MDN says about HTML headings](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements):

> `<h1>` is the highest section level and `<h6>` is the lowest.

which aligns with my understanding, that _higher level_ means _more important_ and _lower level_ means _less important_ (instead of comparing the heading level by numbers).

```suggestion
3. A group may have child groups: Any groups directly following the
   group that have a heading level lower than the group's heading
   are parsed as children.
```

Reply:
Oh, this has a large potential for confusion. `<h1>` is the highest heading level, but it is also the lowest numerically. 

I've rephrased the section and added an explicit explanation mentioning that `<h1>` is to be considered the highest heading level.